File: data/CodeLlama/readable_format_HPEC/fortran/file119.f90
program DRB011_minusminus_orig_yes
    use omp_lib
    implicit none

    integer :: i, len, numNodes, numNodes2
    integer :: x(100)
    len = 100
    numNodes=len
    numNodes2=0

    do i = 1, len
        if (MOD(i,2) == 0) then
            x(i) = 5
        else
            x(i) = -5
        end if
    end do

    !$omp parallel do
    do i = numNodes, 1, -1
        if (x(i) <= 0) then
            numNodes2 = numNodes2-1
        end if
    end do
    !$omp end parallel do

    print*,"numNodes2 =", numNodes2
end program

File: data/gpt-4o/readable_format_HPEC/cpp/file119.cpp
#include <stdio.h>
#include <omp.h>

int main()
{
    int i, len=100, numNodes=len, numNodes2=0;
    int x[100];

    for (i=0; i<len; i++)
    {
        if (i%2 == 0)
            x[i] = 5;
        else
            x[i] = -5;
    }

#pragma omp parallel for
    for (i=numNodes-1; i>=0; i--)
    {
        if (x[i] <= 0)
        {
            #pragma omp atomic
            numNodes2--;
        }
    }

    printf("numNodes2 = %d\n", numNodes2);
    return 0;
}
